cmd/exec.go

```diff
@@ -0,0 +1,196 @@
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/open-policy-agent/opa/cmd/internal/exec"
+	"github.com/open-policy-agent/opa/internal/config"
+	internal_logging "github.com/open-policy-agent/opa/internal/logging"
+	"github.com/open-policy-agent/opa/logging"
+	"github.com/open-policy-agent/opa/plugins"
+	"github.com/open-policy-agent/opa/plugins/bundle"
+	"github.com/open-policy-agent/opa/plugins/discovery"
+	"github.com/open-policy-agent/opa/plugins/logs"
+	"github.com/open-policy-agent/opa/plugins/status"
+	"github.com/open-policy-agent/opa/sdk"
+	"github.com/open-policy-agent/opa/util"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+
+	params := &exec.Params{
+		Output:       os.Stdout,
+		OutputFormat: util.NewEnumFlag("pretty", []string{"pretty", "json"}),
+		LogLevel:     util.NewEnumFlag("error", []string{"debug", "info", "error"}),
+		LogFormat:    util.NewEnumFlag("json", []string{"text", "json", "json-pretty"}),
+	}
+
+	var cmd = &cobra.Command{
+		Use:   `exec <path> [<path> [...]]`,
+		Short: "Execute against input files",
+		Long: `Execute against input files.
+
+The 'exec' command executes OPA against one or more input files. If the paths
+refer to directories, OPA will execute contains files contained inside those
+directories, recursively.
+
+The 'exec' command accepts a --config-file/-c or series of --set options as
+arguments. These options behave the same as way as 'opa run'. Since the 'exec'
+command is intended to execute OPA in a one-shot mode, the 'exec' command will
+manually trigger plugins before and after policy execution:
+
+* Discovery (before)
+* Bundle (before)
+* Status (before)
+* Decision Logs (after)
+
+By default, the 'exec' command executes the "default decision" (specified in
+the OPA configuration) against each input file.`,
+
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return fmt.Errorf("expected at least one path")
+			}
+			return nil
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			params.Paths = args
+			if err := runExec(params); err != nil {
+				fmt.Fprintln(os.Stderr, err)
+				os.Exit(1)
+			}
+		},
+	}
+
+	addOutputFormat(cmd.Flags(), params.OutputFormat)
+	addConfigFileFlag(cmd.Flags(), &params.ConfigFile)
+	addConfigOverrides(cmd.Flags(), &params.ConfigOverrides)
+	addConfigOverrideFiles(cmd.Flags(), &params.ConfigOverrideFiles)
+	cmd.Flags().VarP(params.LogLevel, "log-level", "l", "set log level")
+	cmd.Flags().VarP(params.LogFormat, "log-format", "", "set log format")
```

Comment:
[nit] If there's no short variant, we could do
```suggestion
	cmd.Flags().Var(params.LogFormat, "log-format", "set log format")
```
(IIRC)